src/c/core/communication/serial_protocol.c

```diff
@@ -257,7 +257,7 @@ size_t uxr_read_serial_msg(uxrSerialIO* serial_io,
         }
     }
 
-    if (0 < (bytes_read[0] + bytes_read[1]))
+    if (0 < (bytes_read[0] + bytes_read[1]) || (serial_io->rb_tail != serial_io->rb_head))
```

Comment:
Does exist the case where the first part is false and the second is true? 
If no bytes were read, the head and the tail of the second part have always the same position?

Reply:
Yes, it occurs when you do not read anything from the hardware but you have data in the processing buffer due to the previous hardware reading, that is, when you read two message from a serial reading.

Reply:
ok, great!